File: packages/point/python_std/point_std/path.py
from __future__ import annotations

import os


def _to_point_path(value: str) -> str:
	return value.replace("\\", "/")


def pathJoin(left: str, right: str) -> str:
	return _to_point_path(os.path.join(left, right))


def pathBasename(value: str) -> str:
	return os.path.basename(value)


def pathDirname(value: str) -> str:
	return _to_point_path(os.path.dirname(value))


def pathExtname(value: str) -> str:
	return os.path.splitext(value)[1]


def pathResolve(value: str) -> str:
	return _to_point_path(os.path.abspath(value))


def pathIsAbsolute(value: str) -> bool:
	return os.path.isabs(value)

File: packages/point/src/std/path.ts
import { basename, dirname, extname, isAbsolute, join, resolve } from "node:path";

export function pathJoin(left: string, right: string): string {
	return join(left, right);
}

export function pathBasename(value: string): string {
	return basename(value);
}

export function pathDirname(value: string): string {
	return dirname(value);
}

export function pathExtname(value: string): string {
	return extname(value);
}

export function pathResolve(value: string): string {
	return resolve(value);
}

export function pathIsAbsolute(value: string): boolean {
	return isAbsolute(value);
}
